pkg/controller/iam/serviceaccountkey.go

```diff
@@ -0,0 +1,336 @@
+		}, nil
+	}
+
+	getCall := s.serviceAccountKeyClient.Get(rrn)
+
+	if cr.Spec.ForProvider.PublicKeyType != nil {
+		getCall = getCall.PublicKeyType(*cr.Spec.ForProvider.PublicKeyType)
+	}
+
+	fromProvider, err := getCall.Context(ctx).Do()
+
+	if gcp.IsErrorNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	} else if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetServiceAccountKey)
+	}
+
+	if err := populateSaKey(cr, fromProvider); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetServiceAccountKey)
+	}
+
+	if cr.Status.AtProvider.KeyID != "" {
+		cr.Status.SetConditions(xpv1.Available())
+	}
+
+	connDetails, err := getConnectionDetails(cr.Spec.ForProvider.PublicKeyType, fromProvider)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetServiceAccountKey)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists: true,
+		// all service account key parameters are immutable, no update method exists in Google Cloud API for SA keys
+		ResourceUpToDate:  true,
+		ConnectionDetails: connDetails,
+	}, nil
+}
+
+// Create https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts.keys/create
+func (s *serviceAccountKeyExternalClient) Create(ctx context.Context,
+	mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.ServiceAccountKey)
+
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotServiceAccountKey)
+	}
+
+	cr.SetConditions(xpv1.Creating())
+	// The first parameter to the Create method is the resource name of the GCP service account
+	// which the service account key belongs to. Retry resolution because reference could have been altered
+	// between invocations of Observe and Create
+	saPath, err := s.rrn.referencedServiceAccountPath(cr)
+
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateServiceAccountKey)
+	}
+
+	req := s.serviceAccountKeyClient.Create(saPath, &iamv1.CreateServiceAccountKeyRequest{
+		KeyAlgorithm:   gcp.StringValue(cr.Spec.ForProvider.KeyAlgorithm),
+		PrivateKeyType: gcp.StringValue(cr.Spec.ForProvider.PrivateKeyType),
+	})
+
+	if fromProvider, err := req.Context(ctx).Do(); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateServiceAccountKey)
+	} else if err := populateSaKey(cr, fromProvider); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateServiceAccountKey)
+	} else if connDetails, err := getConnectionDetails(cr.Spec.ForProvider.PublicKeyType, fromProvider); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateServiceAccountKey)
+	} else if keyID, err := parseKeyIDFromRrn(fromProvider.Name); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateServiceAccountKey)
+	} else {
+		meta.SetExternalName(cr, keyID) // set external name to key id parsing it from Google Cloud API relative resource name
```

Comment:
```suggestion
		meta.SetExternalName(cr, path.Base(resourcePath.Path)) // set external name to key id parsing it from Google Cloud API relative resource name
```

The response object schema and format are considered as API contracts so it should be fine to rely on the URL format here.